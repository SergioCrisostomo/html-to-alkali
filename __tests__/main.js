
'use strict';

const tests = require('./assets.json');
const alkali = require('alkali');
const convert = require('../builds/bundle.js').convert;
const Div = alkali.Div;
const Span = alkali.Span;
const P = alkali.P;

const toJSON = (function (attrs) { // make a JSON from a Alkali instance
	return function (obj){
		const tree = attrs.reduce((json, attr) => {
			if (attr == 'children') json.children = Array.from(obj.children).map(ch => toJSON(ch));
			else if (attr == 'textContent' && json.children.length) return json;
			else json[attr] = obj[attr];
			return json;
		}, {});
		return tree;
	}
})(['id', 'className', 'children', 'textContent']);

describe('HTML to Alkali tests', () => {
	it('should convert a valid HTML string', () => {
		tests.parse.forEach(test => {
			expect(convert(test.html)).toEqual(test.js);
		});
	});

	it('should produce valid Alkali API code', () => {
		window.foo = function(){}; // to prevent throw
		tests.parse.forEach(test => {
			const jsString = convert(test.html);
			const Alkali = new (eval(jsString));
			const tree = toJSON(Alkali);
			expect(tree).toMatchSnapshot();
		});
		delete window.foo;
	});
});
