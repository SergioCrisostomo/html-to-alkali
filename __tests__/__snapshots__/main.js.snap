exports[`HTML to Alkali tests should produce valid Alkali API code 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [],
      "className": "",
      "id": "inner",
      "textContent": "Click me"
    }
  ],
  "className": "full red",
  "id": "wrapper"
}
`;

exports[`HTML to Alkali tests should produce valid Alkali API code 2`] = `
Object {
  "children": Array [
    Object {
      "children": Array [],
      "className": "",
      "id": "",
      "textContent": "Hello"
    },
    Object {
      "children": Array [
        Object {
          "children": Array [],
          "className": "inner-span",
          "id": "",
          "textContent": "World"
        }
      ],
      "className": "",
      "id": ""
    }
  ],
  "className": "",
  "id": ""
}
`;

exports[`HTML to Alkali tests should produce valid Alkali API code 3`] = `
Object {
  "children": Array [
    Object {
      "children": Array [],
      "className": "inner-span",
      "id": "",
      "textContent": "Hello"
    },
    Object {
      "children": Array [],
      "className": "inner-span",
      "id": "",
      "textContent": "World"
    }
  ],
  "className": "",
  "id": ""
}
`;
